src/app/pages/review-attributes/review-attributes.page.ts

```diff
@@ -63,65 +56,64 @@ export class ReviewAttributesPage implements OnInit {
 
   onApprovePhraseBtnClick() {
     const self = this;
-
+    const phrase = self._reviewAttributesModelService.get();
+    var reviewId = phrase.id; // get from modelservice
+    var reasonID = 1;
+    self._reviewAttributesModelService.saveReviewAttributes(reviewId,reasonID);
     self._alertService.show({
-      header: "Message approved!",
-      buttons: [
-        {
-          text: "OK",
-          role: "cancel",
-          handler: () => {},
-        },
-      ],
-    });
+    header: 'Message approved!',
+    buttons: [{
+      text: 'OK', role: 'cancel',
+      handler: () => {
+      }
+      }]
+    })
 
     this.phraseToBeReviewed = "";
     this.getNextPhraseButtonDisabled = false;
     this.approveAndRejectButtonDisplayed = false;
   }
 
   onGetNextPhraseBtnClick() {
+
     this.getNextPhraseToBeReviewed();
   }
 
   getNextPhraseToBeReviewed() {
-    const self = this;
-    const phraseTBR =
-      this._reviewAttributesModelService.getNextReviewEligiblePhrase();
-    if (phraseTBR) {
-      const phraseTBRAsString = this.getAttrString(phraseTBR);
+    this._reviewAttributesModelService.getNewPhrase();
+    const wait = ms => new Promise(resolve => setTimeout(resolve, ms));
+    console.log("newphrase loading ...");
+    wait(1000).then(() => {
+      const phraseTBRAsString = this.getAttrString(this._reviewAttributesModelService.get()); 
+      console.log(phraseTBRAsString);
       this.phraseToBeReviewed = phraseTBRAsString;
       this.getNextPhraseButtonDisabled = true;
       this.approveAndRejectButtonDisplayed = true;
-    } else {
-      self._alertService.show({
-        header: "No more phrases to review!",
-        buttons: [
-          {
-            text: "OK",
-            role: "cancel",
-            handler: () => {},
-          },
-        ],
-      });
-    }
+    });
   }
 
   getAttrString(tbr) {
     let rtn = "";
 
-    if (tbr.adverb) rtn += tbr.adverb + " ";
+    if (tbr.adverb)
+      rtn += tbr.adverb + " ";
 
     rtn += tbr.verb + " ";
 
-    if (tbr.preposition) rtn += tbr.preposition + " ";
+    if (tbr.preposition)
+      rtn += tbr.preposition + " ";
 
     rtn += tbr.noun;
 
     return rtn;
   }
 
   getListOfReviewDecisionReasons() {
-    return;
+    const reasons = [
```

Comment:
This needs to come from the backend. I notice there is not an endpoint for it yet (on the feature/review-attributes-page branch), so I created TRIB-166. If you're up for a backend issue, you can knock that out, or we can get it scheduled otherwise. You let me know.

Reply:
Note that this is still an issue.. we should be calling the endpoint defined in TRIB-166 to get the list of reasons.